src/Sprites/ArtFile.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include "../Rect.h"
+#include <array>
+#include <vector>
+#include <cstdint>
+
+struct Color {
+	Color();
+
+	uint8_t red;
+	uint8_t green;
+	uint8_t blue;
+	uint8_t alpha;
```

Comment:
I believe the newer C++ standards allow `= 0` initializers in struct definitions. That might be a good alternative to supplying an explicit constructor.

Reply:
I like this idea and will try it out. It is a cleaner approach.

Reply:
FYI: On a hunch, I just did a bit of testing. It seems defaulting fields to some value in a struct definition is the same as providing an explicit constructor. That is, the type is no longer an aggregate, and you'll get errors from brace initialization of instance variables.